src/Native/Unix/System.Native/pal_networking.c

```diff
@@ -1654,6 +1654,22 @@ static bool TryGetPlatformSocketOption(int32_t socketOptionName, int32_t socketO
 
                 // case SocketOptionName_SO_TCP_BSDURGENT:
 
+                case SocketOptionName_SO_TCP_KEEPALIVE_RETRYCOUNT:
+                    *optName = TCP_KEEPCNT;
+                    return true;
+
+                case SocketOptionName_SO_TCP_KEEPALIVE_TIME:
+                    #ifdef OS_MACOSX
+                    *optName = TCP_KEEPALIVE;
+                    #else
+                    *optName = TCP_KEEPIDLE;
+                    #endif
```

Comment:
We prefer to use cmake configuration for things like this, e.g. https://github.com/luigiberrettini/corefx/blob/e2b75cc0c82619e61ee43418f9be580f9342aec1/src/Native/Unix/configure.cmake#L546